ci20.c

```diff
@@ -0,0 +1,562 @@
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+};
+
+/* SPI Bus Parameters */
+static uint8_t     spiMode   = 0;
+static uint8_t     spiBPW    = 8;
+static uint16_t    spiDelay  = 0;
+static uint32_t    spiSpeeds[2];
+static int         spiFds[2];
+
+int ci20ValidGPIO(int pin) {
+	if(pinToGpio[pin] != -1) {
+		return 0;
+	}
+	return -1;
+}
+
+static int changeOwner(char *file) {
+	uid_t uid = getuid();
+	uid_t gid = getgid();
+
+	if(chown(file, uid, gid) != 0) {
+		if(errno == ENOENT)	{
+			wiringXLog(LOG_ERR, "ci20->changeOwner: File not present: %s", file);
+			return -1;
+		} else {
+			wiringXLog(LOG_ERR, "ci20->changeOwner: Unable to change ownership of %s: %s", file, strerror (errno));
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+static int identify(void) {
+	FILE *cpuFd;
+	char line[120], revision[120], hardware[120], name[120];
+
+	if((cpuFd = fopen("/proc/cpuinfo", "r")) == NULL) {
+		wiringXLog(LOG_ERR, "ci20->identify: Unable open /proc/cpuinfo");
+		return -1;
+	}
+
+	while(fgets(line, 120, cpuFd) != NULL) {
+		if(strncmp(line, "Revision", 8) == 0) {
+			strcpy(revision, line);
+		}
+		if(strncmp(line, "Hardware", 8) == 0) {
+			strcpy(hardware, line);
+		}
+	}
+
+	fclose(cpuFd);
+
+	sscanf(hardware, "Hardware%*[ \t]:%*[ ]%[a-zA-Z0-9 ./()]%*[\n]", name);
+
+	if(strstr(name, "CI20") != NULL) {
+		return 0;
+	} else {
+		return -1;
+	}
+}
+
+static int setup(void) {
+
+	return 0;
+}
+
+static int ci20DigitalRead(int pin) {
+	char path[PATH_MAX];
+	char value_str[3];
+	int fd;
+
+	if(pinModes[pin] != INPUT && pinModes[pin] != SYS) {
+		wiringXLog(LOG_ERR, "ci20->digitalRead: Trying to write to pin %d, but it's not configured as input", pin);
+		return -1;
+	}
+
+	if(ci20ValidGPIO(pin) != 0) {
+		wiringXLog(LOG_ERR, "ci20->digitalRead: Invalid pin number %d (0 >= pin <= 31)", pin);
+		return -1;
+	}
+	
+	snprintf(path, PATH_MAX, "/sys/class/gpio/gpio%d/value", pinToGpio[pin]);
+	fd = open(path, O_RDONLY);
+	if (fd == -1) {
+		wiringXLog(LOG_ERR, "ci20->digitalRead: Failed to open gpio value for reading", pin);
+		return -1;
+	}
+
+	if (read(fd, value_str, 1) == -1) {
+		wiringXLog(LOG_ERR, "ci20->digitalRead: Failed to read value", pin);
+		return -1;
+	}
+
+	close(fd);
+
+	if(value_str[0]!='0') {
+		return HIGH;
+	} else {
+		return LOW;
+	}
+}
+
+static int ci20DigitalWrite(int pin, int value) {
+	static const char s_values_str[] = "01";
+	char path[PATH_MAX];
+	int fd;
+	
+	if(pinModes[pin] != OUTPUT) {
+		wiringXLog(LOG_ERR, "ci20->digitalWrite: Trying to write to pin %d, but it's not configured as output", pin);
+		return -1;
+	}
+
+	if(ci20ValidGPIO(pin) != 0) {
+		wiringXLog(LOG_ERR, "ci20->digitalWrite: Invalid pin number %d (0 >= pin <= 31)", pin);
+		return -1;
+	}	
+	
+	snprintf(path, PATH_MAX, "/sys/class/gpio/gpio%d/value", pinToGpio[pin]);
+	fd = open(path, O_WRONLY);
+	if (fd == -1) {
```

Comment:
Can you make sure the coding style is consistent between all modules:
`if (` > `if(`